### PR TITLE
[4.0] Show only the new-button if there are no items.

### PR DESCRIPTION
**Pull request by chmst**
Many list views in backend can be empty. As for example articles, menu items, contacts ....
Nevertheless there is always a full set of action buttons: publish, unpublish, batch, trash. But none of these has a function as long as the list is empty.
So only the one button "new" is needed, all the other buttons are useless.
This patch shows for the menu-items view how it could be made easier in a very simple way. 
It is also a step towards accessibility as there are less elements to be tabbed or read by a screen reader. 

This could be made for many list views: articles, contacts, user notes and so on if this is acepted.

### Testing Instructions
Have a look on a empty Menu items list

### Actual result
empty-list-view-before

### After Patch
empty-list-view-after

### Documentation Changes Required
yes

### CODE REVIEW

File: administrator/components/com_menus/View/Items/HtmlView.php [hunk 1]
```diff
@@ -341,53 +341,56 @@ protected function addToolbar()
 		{
 			ToolbarHelper::addNew('item.add');
 		}
-
-		$protected = $this->state->get('filter.menutype') == 'main';
-
-		if ($canDo->get('core.edit.state') && !$protected)
+
```
Comment (Quy) [inline administrator/components/com_menus/View/Items/HtmlView.php]: Remove tabs.

File: administrator/components/com_menus/View/Items/HtmlView.php [hunk 2]
```diff
@@ -341,53 +341,56 @@ protected function addToolbar()
...
+			if (!$protected && $this->state->get('filter.published') == -2 && $canDo->get('core.delete'))
+			{
+				ToolbarHelper::deleteList('JGLOBAL_CONFIRM_DELETE', 'items.delete', 'JTOOLBAR_EMPTY_TRASH');
+			}
+			elseif (!$protected && $canDo->get('core.edit.state'))
+			{
+				ToolbarHelper::trash('items.trash');
+			}
+		}
+
```
Comment (Quy) [inline administrator/components/com_menus/View/Items/HtmlView.php]: Remove tabs

**Comment by brianteeman** [comment]
With the proposed new UI and a combined actions button is this even needed

**Comment by carlitorweb** [comment]
We need to extend this for all components for sure

**Comment by carlitorweb** [comment]
Tested here ok. Waiting for the remove of whitespace tabs

**Comment by carlitorweb** [comment]
@Quy a question about this, do you think ToolbarHelper class can have a param in each method (where it makes sense to check if there are elements for actions like publish, delete etc) for check this? In this way when you do ToolbarHelper::publish(....) he will check first if the actual view has items where these action have sense.

**Comment by brianteeman** [comment]
please dont misunderstand me - i am a massive fan of progressive disclosure of buttons and I have tried with others to add this to joomla for 4 years. But in the new proposed toolbar there will just be a new and an actions button so I dont believe it is needed anymore

**Comment by carlitorweb** [comment]
@brianteeman really? where I can see this you said?

**Comment by brianteeman** [comment]
quick example https://github.com/joomla/40-backend-template/issues/422

**Comment by carlitorweb** [comment]
Thank you @brianteeman . With a fast look, still relevant this PR IMO, is a button and a dropdown less

File: administrator/components/com_menus/View/Items/HtmlView.php (hunk 1 above)
Comment (chmst) [reply administrator/components/com_menus/View/Items/HtmlView.php]: Sorry, I cannot see where the tabs come from. In my IDE all looks fine. Can you help me with that?

Comment (Quy) [reply administrator/components/com_menus/View/Items/HtmlView.php]: See  line 344 where tabs highlighted in blue.

tabs

**Comment by chmst** [comment]
@Quy thank you - I've commited the changes.

**Comment by chmst** [comment]
Now as I have seen the new Screen, I think that also Elements like Search tools can be hidden on a empty list, because they cannot give a result. This could be made with the same simple condition.

**Comment by mbabker** [comment]
> I think that also Elements like Search tools can be hidden on a empty list, because they cannot give a result

You can't hide search tools on an empty screen.  If you search/filter for something that yields no search results, you'd hide the search bar and make it impossible to clear the filters without logging out.  This bug already exists in core (in fact we fixed one of those with 3.8.8).

**Comment by chmst** [comment]
@mbabker you are right - if filters are not empty, it's another use case.

**Comment by Quy** [comment]
`Rebuild` is an independent action from the other buttons. Should it be hidden too?

**Comment by chmst** [comment]
So it seems that it is better to ask for each or element if it makes sense. If there are no items at all, it is not necessary. So @carlitorweb has already mentioned the ToolbarHelper Class, which could be used for this purpose.

> @Quy a question about this, do you think ToolbarHelper class can have a param in each method (where it makes sense to check if there are elements for actions like publish, delete etc) for check this? In this way when you do ToolbarHelper::publish(....) he will check first if the actual view has items where these action have sense.

**Comment by chmst** [comment]
Closed as there are other solutions

**Pull request closed without merging**